deps/rabbitmq_cli/lib/rabbitmq/cli/diagnostics/commands/check_if_any_deprecated_features_are_used_command.ex

```diff
@@ -0,0 +1,65 @@
+## This Source Code Form is subject to the terms of the Mozilla Public
+## License, v. 2.0. If a copy of the MPL was not distributed with this
+## file, You can obtain one at https://mozilla.org/MPL/2.0/.
+##
+## Copyright (c) 2007-2023 VMware, Inc. or its affiliates.  All rights reserved.
+
+defmodule RabbitMQ.CLI.Diagnostics.Commands.CheckIfAnyDeprecatedFeaturesAreUsedCommand do
+  alias RabbitMQ.CLI.Core.DocGuide
+
+  alias RabbitMQ.CLI.Diagnostics.Commands.{
+    CheckIfClusterHasClassicQueueMirroringPolicyCommand
+  }
+
+  @behaviour RabbitMQ.CLI.CommandBehaviour
+
+  def scopes(), do: [:ctl, :diagnostics]
+
+  use RabbitMQ.CLI.Core.MergesNoDefaults
+
+  def validate([_ | _] = args, _) when length(args) != 0,
+    do: {:validation_failure, :too_many_args}
+
+  def validate([], %{formatter: formatter}) do
+    case formatter do
+      "report" -> :ok
+      _other -> {:validation_failure, "Only report formatter is supported"}
+    end
+  end
+
+  def validate([], _), do: :ok
+
+  use RabbitMQ.CLI.Core.RequiresRabbitAppRunning
+
+  def run([], %{node: node_name} = opts) do
+        [
+          run_command(CheckIfClusterHasClassicQueueMirroringPolicyCommand, [], opts)
```

Comment:
This approach will be very painful to format correctly, and the `reporter` formatter is something highly non-standard that we should not try to replicate.

Instead, the health check function on the node should return an easy to format error and a list of deprecated features that are used. Then this command will simply format a list.

Reply:
For example, a standard test in CLI tools is to check that when a node is down, a reasonable error is returned.

This "compound command" does not handle `:badrpc` results, in fact, it wraps them in a way that's difficult to match on (it's a nested tuple).

Reply:
I just rewrote the command to just return the list of deprecated features